Move ADC API test to use dt_spec

benjaminbjornsson opened the pull request:
Some tests are quite verbose due to lengthy if-defs and defines for information that can be moved to the devicetree.

Made this a draft to allow for input early on in the process.

benjaminbjornsson committed:
tests: drivers: adc: adc_api: Move test to devicetree

Move test to use adc_dt_spec for native posix board.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

benjaminbjornsson commented:
@erwango Would you mind having a look to make sure I'm on the right track here?

benjaminbjornsson commented:
There'll be an awful lot of overlay files... Is there a better way?

erwango reviewed (commented):
Didn't look at details, but heavy +1 on the direction.
Thanks for taking a stab at this !

benjaminbjornsson committed:
tests: drivers: adc: adc_api: Move compat infineon,xmc4xxx-adc to DT

Move channel description to devicetree for infineon,xmc4xxx-adc
compatible to make the test more readable.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

benjaminbjornsson committed:
tests: drivers: adc: adc_api: Move compat nxp_mcux,12b1msps-sar to DT

Move channel description to devicetree for nxp_mcux,12b1msps-sar
compatible to make the test more readable.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

benjaminbjornsson committed:
tests: drivers: adc: adc_api: Move compat ite,it8xxx2-adc to DT

Move channel description to devicetree for ite,it8xxx2-adc
compatible to make the test more readable.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

benjaminbjornsson committed:
tests: drivers: adc: adc_api: Move compat ti,cc13xx-cc26xx-adc to DT

Move channel description to devicetree for ti,cc13xx-cc26xx-adc
compatible to make the test more readable.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

benjaminbjornsson committed:
tests: drivers: adc: adc_api: Move compat ti,cc32xx-adc to DT

Move channel description to devicetree for ti,cc32xx-adc
compatible to make the test more readable.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

benjaminbjornsson committed:
tests: drivers: adc: adc_api: Move compat nuvoton,npcx-adc to DT

Move channel description to devicetree for nuvoton,npcx-adc
compatible to make the test more readable.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

benjaminbjornsson committed:
tests: drivers: adc: adc_api: Move compat nxp,lpc-lpadc to DT

Move channel description to devicetree for nxp,lpc-lpadc
compatible to make the test more readable.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

benjaminbjornsson committed:
tests: drivers: adc: adc_api: Move compat microchip,xec-adc to DT

Move channel description to devicetree for microchip,xec-adc
compatible to make the test more readable.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

benjaminbjornsson committed:
tests: drivers: adc: adc_api: Move compat nxp,kinetis-adc12 to DT

Move channel description to devicetree for nxp,kinetis-adc12
compatible to make the test more readable.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

benjaminbjornsson committed:
tests: drivers: adc: adc_api: Move compat st,stm32-adc to DT

Move channel description to devicetree for st,stm32-adc
compatible to make the test more readable.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

benjaminbjornsson committed:
tests: drivers: adc: adc_api: Move compat nordic,nrf-adc to DT

Move channel description to devicetree for nordic,nrf-adc
compatible to make the test more readable.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

benjaminbjornsson committed:
tests: drivers: adc: adc_api: Move compat nordic,nrf-saadc to DT

Move channel description to devicetree for nordic,nrf-saadc
compatible to make the test more readable.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

benjaminbjornsson committed:
tests: drivers: adc: adc_api: Move compat nxp,kinetis-adc16 to DT

Move channel description to devicetree for nxp,kinetis-adc16
compatible to make the test more readable.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

benjaminbjornsson committed:
tests: drivers: adc: adc_api: Move compat telink,b91-adc to DT

Move channel description to devicetree for telink,b91-adc
compatible to make the test more readable.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

benjaminbjornsson committed:
tests: drivers: adc: adc_api: Move compat gd,gd32-adc to DT

Move channel description to devicetree for gd,gd32-adc
compatible to make the test more readable.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

benjaminbjornsson committed:
tests: drivers: adc: adc_api: Move compat atmel,sam-afec to DT

Move channel description to devicetree for atmel,sam-afec
compatible to make the test more readable.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

benjaminbjornsson committed:
tests: drivers: adc: adc_api: Move compat atmel,sam-adc to DT

Move channel description to devicetree for atmel,sam-adc
compatible to make the test more readable.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

benjaminbjornsson committed:
tests: drivers: adc: adc_api: Move compat atmel,sam0-adc to DT

Move channel description to devicetree for atmel,sam0-adc
compatible to make the test more readable.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

benjaminbjornsson committed:
tests: drivers: adc: adc_api: Remove mikroe adc click sheild

Remove shield, using this shild will require an overlay for
the particular board being used.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

benjaminbjornsson committed:
tests: drivers: adc: adc_api: Move compat espressif,esp32-adc to DT

Move channel description to devicetree for espressif,esp32-adc
compatible to make the test more readable.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

benjaminbjornsson committed:
tests: drivers: adc: adc_api: Move compat raspberrypi,pico-adc to DT

Move channel description to devicetree for raspberrypi,pico-adc
compatible to make the test more readable.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

erwango commented on tests/drivers/adc/adc_api/boards/96b_aerocore2.overlay:
```diff
@@ -0,0 +1,27 @@
+ * Copyright (c) Benjamin Björnsson <benjamin.bjornsson@gmail.com>
+ */
+
+#include <zephyr/dt-bindings/adc/adc.h>
```
Nitpick (not blocking): This include should already be done in stm32xxxx.dtsi files

benjaminbjornsson replied:
Tried my best to remove those not needed, include is missing for C0, F4, H5 and U5.

benjaminbjornsson replied:
I can take care of this, the problem exists for more than just the SoCs from ST named above.

erwango commented on tests/drivers/adc/adc_api/boards/stm32h573i_dk.overlay:
```diff
@@ -1,9 +1,28 @@
+};
+
 &adc1 {
-	/delete-property/ pinctrl-0;
-	pinctrl-0 = <&adc1_inp0_pa0>; /* Arduino A2 */
```
This is likely still required.

benjaminbjornsson replied:
I may have misunderstood this, but this pinctrl should be paired with channel 0 right?

I focused on moving the channel settings as is for clarity, but I realized that lots of boards have incorrect or missing pinctrls. I’m looking at correcting this in a future PR.

erwango reviewed (commented):
Otherwise LGTM.
I only looked at STM32 bits

benjaminbjornsson commented on tests/drivers/adc/adc_api/src/test_adc.c:
```diff
@@ -450,61 +21,42 @@
-#endif /* defined(ADC_2ND_CHANNEL_ID) */
+	for (int i = 0; i < adc_channels_count; i++) {
+		ret = adc_channel_setup_dt(&adc_channels[i]);
+		zassert_equal(ret, 0, "Setting up of the first channel failed with code %d", ret);
```
Fixed.

benjaminbjornsson commented on tests/drivers/adc/adc_api/src/test_adc.c:
```diff
@@ -560,42 +107,38 @@ ZTEST_USER(adc_basic, test_adc_sample_one_channel)
-		return TC_FAIL;
+	for (int i = 0; i < adc_channels_count; i++) {
+		sequence.channels |= BIT(adc_channels[i].channel_id);
 	}
```
Agreed and fixed.

benjaminbjornsson commented on tests/drivers/adc/adc_api/boards/xmc45_relax_kit.overlay:
```diff
@@ -1,3 +1,29 @@
+};
+
 &adc0 {
-	status = "okay";
```
Nice catch, thanks!

benjaminbjornsson commented on tests/drivers/adc/adc_api/boards/frdm_k22f.overlay:
```diff
@@ -0,0 +1,31 @@
+};
+
+&adc0 {
+	status = "okay";
```
My misstake, thanks!

anangl commented on tests/drivers/adc/adc_api/src/test_adc.c:
```diff
@@ -764,26 +291,20 @@ static int test_task_repeated_samplings(void)
 
-	if (!adc_dev) {
-		return TC_FAIL;
+	if (adc_channels_count > 1) {
+		sequence.channels |=  BIT(adc_channels[1].channel_id);
```
Just for consistency, it would be better to also here use:
```suggestion
	for (int i = 1; i < adc_channels_count; i++) {
		sequence.channels |= BIT(adc_channels[i].channel_id);
```
Then it should be possible to use even more than 2 channels for a board, provided that the supplied buffer is big enough (but that is probably beyond the scope of this PR).

benjaminbjornsson replied:
Agreed, wilL keep this in mind as I try to improvE this test in a future PR.

anangl reviewed (approved):
Great job. Thanks for taking this effort!
